c_glib/arrow-glib/array-builder.h

```diff
@@ -45,6 +45,59 @@ GArrowArray        *garrow_array_builder_finish   (GArrowArrayBuilder *builder,
                                                    GError **error);
 
 
+#define GARROW_TYPE_NULL_ARRAY_BUILDER       \
+  (garrow_null_array_builder_get_type())
+#define GARROW_NULL_ARRAY_BUILDER(obj)                                  \
+  (G_TYPE_CHECK_INSTANCE_CAST((obj),                                    \
+                              GARROW_TYPE_NULL_ARRAY_BUILDER,           \
+                              GArrowNullArrayBuilder))
+#define GARROW_NULL_ARRAY_BUILDER_CLASS(klass)                  \
+  (G_TYPE_CHECK_CLASS_CAST((klass),                             \
+                           GARROW_TYPE_NULL_ARRAY_BUILDER,      \
+                           GArrowNullArrayBuilderClass))
+#define GARROW_IS_NULL_ARRAY_BUILDER(obj)                               \
+  (G_TYPE_CHECK_INSTANCE_TYPE((obj),                                    \
+                              GARROW_TYPE_NULL_ARRAY_BUILDER))
+#define GARROW_IS_NULL_ARRAY_BUILDER_CLASS(klass)               \
+  (G_TYPE_CHECK_CLASS_TYPE((klass),                             \
+                           GARROW_TYPE_NULL_ARRAY_BUILDER))
+#define GARROW_NULL_ARRAY_BUILDER_GET_CLASS(obj)                \
+  (G_TYPE_INSTANCE_GET_CLASS((obj),                             \
+                             GARROW_TYPE_NULL_ARRAY_BUILDER,    \
+                             GArrowNullArrayBuilderClass))
+
+typedef struct _GArrowNullArrayBuilder         GArrowNullArrayBuilder;
+typedef struct _GArrowNullArrayBuilderClass    GArrowNullArrayBuilderClass;
+
+/**
+ * GArrowNullArrayBuilder:
+ *
+ * It wraps `arrow::NullBuilder`.
+ */
+struct _GArrowNullArrayBuilder
+{
+  /*< private >*/
+  GArrowArrayBuilder parent_instance;
+};
+
+struct _GArrowNullArrayBuilderClass
+{
+  GArrowArrayBuilderClass parent_class;
+};
+
+GType garrow_null_array_builder_get_type(void) G_GNUC_CONST;
```

Comment:
Could you use `G_DECLARE_DERIVABLE_TYPE`?

```c
#define GARROW_TYPE_NULL_ARRAY_BUILDER (garrow_null_array_builder_get_type())
G_DECLARE_DERIVABLE_TYPE(GArrowNullArrayBuilder,
                         garrow_null_array_builder,
                         GARROW,
                         NULL_ARRAY_BUILDER,
                         GARROW_ARRAY_BUILDER)
struct _GArrowNullArrayBuilderClass
{
  GArrowArrayBuilderClass parent_class;
};
```

Reply:
G_DECLARE_DERIVABLE_TYPE macro is currently used only for GArrowArrayBuilder, GArrowUIntArrayBuilder, and GArrowDecimal128ArrayBuilder.
If this macro can be applied for the other array builders, I'd like to make the new pull-request to apply them the macro.

Reply:
Yes, please.

Reply:
OK, I’ll make it tonight.

Reply:
OK, I’ll make it tonight.

Reply:
This is it https://github.com/apache/arrow/pull/3945